rust/core-lib/src/layers.rs

```diff
@@ -40,8 +41,12 @@ pub struct Scopes {
 pub struct ScopeLayer {
     stack_lookup: Vec<Vec<Scope>>,
     style_lookup: Vec<Style>,
+    // TODO: this might be efficient (in memory at least) if we use
+    // a prefix tree.
+    /// style state of existing scope spans, so we can more efficiently
+    /// compute styles of child spans.
+    style_cache: HashMap<Vec<Scope>, StyleModifier>,
```

Comment:
If hashing shows up any significant amount in the time profile, you can switch this to use a FNV hasher, which is what I use in syntect.

Reply:
@trishume okay, good note, thanks!